08-compiler-features.tex

```diff
@@ -36,6 +139,143 @@ \section{Dead Code Elimination}
+
+\begin{lstlisting}
+import haxe.|
+Will return :
+
+<list>
+    <i n="BaseCode"><t></t><d></d></i>
+    <i n="EnumFlags"><t></t><d></d></i>
+    ...
+    <i n="io"><t></t><d></d></i>
+    <i n="macro"><t></t><d></d></i>
+    <i n="remoting"><t></t><d></d></i>
+    ...
+</list>
+\end{lstlisting}
+
+\paragraph{Error Handling}
+When run in completion mode, the compiler will not display errors but instead try to ignore them or recover from them. 
+
+In case a not recoverable error occur while getting completion, the Haxe compiler will print the error message instead of the completion output. You can then treat any not-XML output as an error message which prevent the completion from being performed.
+
+\paragraph{Compilation cache server}
+\since{2.09}
+
+To get best speed for both compilation and completion, you can use the \ic{--wait} commandline parameter to start a Haxe compilation server. You can also use \ic{-v} to have the server printing the log. Here's an example:
+
+\begin{lstlisting}
+haxe -v --wait 6000
+\end{lstlisting}
+
+You can then connect to the Haxe server, send commandline parameters followed by a 0 byte, then read the response (either completion result or errors).
+
+You can use the \ic{--connect} commandline parameter to have haxe send its compilation commands to the server instead of executing them directly :
+
+\begin{lstlisting}
+haxe --connect 6000 myproject.hxml
+\end{lstlisting}
+
+Please notice that you can use \ic{--cwd} as the first sent commandline parameter in order to change the Haxe server current working directory, since usually class paths and other files are relative to your project.
+
+\paragraph{How it works}
+The compilation server will cache the following things:
+
+\begin{description}
+	\item[parsed files] the files will only get parsed again if they are modified, or if there was a parse error
+	\item[haxelib calls] the previous results of haxelib calls will be reused (only for completion : they are ignored when doing a compilation)
+	\item[typed modules] compilation modules will be cached after a successful compilation, and can be reused in later compilation/completions if none of their dependencies have been modified
+\end{description}
+
+You can get precise timing of the times spent by the compiler and how using the compilation server affects it by adding the \ic{--times} command line.
+
+\paragraph{Protocol}
+As the following Haxe/Neko example show, you can simply connect on the server port and send all commands (one per line) ending with a 0 binary char. You can then read the results.
+
+Macros and other commands can log some data which are not errors. In command line we can make the difference between what is printed on stdout and what is print on stderr, but this is not the case in socket mode. In order to differentiate between the two, log messages (not errors) are prefixed with a \ic{\\x01} character, and all newlines in the message are replaced by the same \ic{\\x01} character.
```

Comment:
"log some events..." (data is singular, so no use of "are not errors"). "In the command line, ..." Review "make the difference" are you making the difference (causing it) or seeing the difference (perceiving it)?, No comma before "but", No comma before "and" Review "newlines" consider: "new lines"